Source/WebKit/Shared/SessionState.cpp

```diff
@@ -30,6 +30,97 @@
+#endif
+    Vector<Ref<FrameState>>&& children, Vector<AtomString>&& documentState
+) {
+    return adoptRef(*new FrameState(urlString, originalURLString, referrer, target, frameID, stateObjectData, documentSequenceNumber, itemSequenceNumber, scrollPosition, shouldRestoreScrollPosition, pageScaleFactor, WTFMove(httpBody), identifier, hasCachedPage, title, shouldOpenExternalURLsPolicy, WTFMove(sessionStateObject), wasCreatedByJSWithoutUserInteraction,
+#if PLATFORM(IOS_FAMILY)
+        exposedContentRect, unobscuredContentRect, minimumLayoutSizeInScrollViewCoordinates, contentSize, scaleIsInitial, obscuredInsets,
+#endif
+        WTFMove(children), WTFMove(documentState)
+    ));
+}
+
+FrameState::FrameState(const String& urlString, const String& originalURLString, const String& referrer, const AtomString& target, std::optional<WebCore::FrameIdentifier> frameID, std::optional<Vector<uint8_t>> stateObjectData, int64_t documentSequenceNumber, int64_t itemSequenceNumber, WebCore::IntPoint scrollPosition, bool shouldRestoreScrollPosition, float pageScaleFactor, std::optional<HTTPBody>&& httpBody, WebCore::BackForwardItemIdentifier identifier, bool hasCachedPage, const String& title, WebCore::ShouldOpenExternalURLsPolicy shouldOpenExternalURLsPolicy, RefPtr<WebCore::SerializedScriptValue>&& sessionStateObject, bool wasCreatedByJSWithoutUserInteraction,
+#if PLATFORM(IOS_FAMILY)
+    WebCore::FloatRect exposedContentRect, WebCore::IntRect unobscuredContentRect, WebCore::FloatSize minimumLayoutSizeInScrollViewCoordinates, WebCore::IntSize contentSize, bool scaleIsInitial, WebCore::FloatBoxExtent obscuredInsets,
+#endif
+    Vector<Ref<FrameState>>&& children, Vector<AtomString>&& documentState
+)
+    : urlString(urlString)
+    , originalURLString(originalURLString)
+    , referrer(referrer)
+    , target(target)
+    , frameID(frameID)
+    , stateObjectData(stateObjectData)
+    , documentSequenceNumber(documentSequenceNumber)
+    , itemSequenceNumber(itemSequenceNumber)
+    , scrollPosition(scrollPosition)
+    , shouldRestoreScrollPosition(shouldRestoreScrollPosition)
+    , pageScaleFactor(pageScaleFactor)
+    , httpBody(WTFMove(httpBody))
+    , identifier(identifier)
+    , hasCachedPage(hasCachedPage)
+    , title(title)
+    , shouldOpenExternalURLsPolicy(shouldOpenExternalURLsPolicy)
+    , sessionStateObject(WTFMove(sessionStateObject))
+    , wasCreatedByJSWithoutUserInteraction(wasCreatedByJSWithoutUserInteraction)
+#if PLATFORM(IOS_FAMILY)
+    , exposedContentRect(exposedContentRect)
+    , unobscuredContentRect(unobscuredContentRect)
+    , minimumLayoutSizeInScrollViewCoordinates(minimumLayoutSizeInScrollViewCoordinates)
+    , contentSize(contentSize)
+    , scaleIsInitial(scaleIsInitial)
+    , obscuredInsets(obscuredInsets)
+#endif
+    , children(WTFMove(children))
+    , m_documentState(WTFMove(documentState))
+{
+}
+
+Ref<FrameState> FrameState::copy()
+{
+    Ref frameState = FrameState::create();
```

Comment:
This should use the version with all the parameters so it will fail to compile if we forget one.